src/containers/asyncContainer.js

```diff
@@ -54,12 +54,16 @@ const asyncContainer = (Typeahead) => {
       // the process of searching. The logic for whether or not to display the
       // custom menu option is basically the same as whether we display the
       // empty label, so use that as a proxy.
-      const shouldAllowNew = allowNew && emptyLabel === props.emptyLabel;
+      const shouldAllowNew =
```

Comment:
The logic feels a little confusing to me. It might be clearer written as:

```jsx
const allowNew = typeof props.allowNew === 'function' ?
  props.allowNew :
  props.allowNew && emptyLabel === props.emptyLabel;
```
or

```jsx
let allowNew = props.allowNew && emptyLabel === props.emptyLabel;
if (typeof props.allowNew === 'function') {
  allowNew = props.allowNew;
}
```

Reply:
Great point. I went with the second approach for readability.